src/components/ui/Button.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+  alignItems,
+  backgroundClip,
+  backgroundColor,
+  backgroundImage,
+  borderColor,
+  borderRadius,
+  borderWidth,
+  boxShadow,
+  boxShadowColor,
+  brightness,
+  classnames,
+  cursor,
+  display,
+  flexDirection,
+  fontFamily,
+  fontSize,
+  fontWeight,
+  gradientColorStops,
+  justifyContent,
+  lineHeight,
+  opacity,
+  outlineStyle,
+  padding,
+  space,
+  textAlign,
+  textColor,
+  transitionDuration,
+  transitionProperty,
+  transitionTimingFunction,
+  width,
+} from '@/classnames/tailwind'
+import Arrow from '@/icons/Arrow'
+import Loading from '@/icons/Loading'
+import React from 'react'
+
+const commonClasses = ({
+  type,
+  fullWidth,
+  center,
+  available,
+  small,
+  gradientFont,
+}: {
+  type: ButtonType
+  fullWidth?: boolean
+  center?: boolean
+  available?: boolean
+  small?: boolean
+  gradientFont?: boolean
+}) =>
+  classnames(
+    display('flex'),
+    flexDirection('flex-row'),
+    justifyContent({ 'justify-center': center }),
+    alignItems('items-center'),
+    fontWeight('font-bold'),
+    fontFamily('font-primary'),
+    transitionProperty('transition-all'),
+    transitionTimingFunction('ease-in-out'),
+    transitionDuration('duration-100'),
+    cursor({ 'cursor-not-allowed': !available }),
+    outlineStyle('focus:outline-none'),
+    opacity({ 'opacity-50': !available }),
+    boxShadow({
+      'shadow-2xl': available,
+      'hover:shadow-lg': available,
+      'active:shadow-button-active': available && !gradientFont,
+    }),
+    width({ 'w-full': fullWidth }),
+    textAlign({ 'text-center': center }),
+    fontSize(small ? 'text-sm' : 'text-lg'),
+    lineHeight(small ? 'leading-5' : 'leading-7'),
+    type !== 'tertiary'
+      ? padding(
+          small
+            ? { 'py-2': true, 'px-4': true }
+            : { 'py-4': true, 'px-6': true }
+        )
+      : undefined,
+    space('space-x-2')
+  )
+
+const button = ({
+  fullWidth,
+  available,
+  center,
+  type,
+  small,
+  gradientFont,
+}: ButtonProps & { available?: boolean }) =>
+  classnames(
+    commonClasses({ type, fullWidth, center, available, small, gradientFont }),
+    colorClasses(type, available, gradientFont)
+  )
+
+const colorClasses = (
+  type: ButtonType,
+  available?: boolean,
+  gradientFont?: boolean
+) => {
+  const hasNoGradient = !gradientFont
+
+  switch (type) {
+    case 'primary':
+      return classnames(
```

Comment:
How about wrapping switch-case into `classnames(switch() { case '': ... }`?

Reply:
it doesn't work, I tried